NorthstarDLL/rpakfilesystem.cpp

```diff
@@ -258,30 +258,64 @@ void*, __fastcall, (const char* pPath, void* pCallback))
 // clang-format on
 {
 	fs::path path(pPath);
-	char* allocatedNewPath = nullptr;
+	std::string newPath = "";
+	fs::path filename = path.filename();
 
 	if (path.extension() == ".stbsp")
 	{
-		fs::path filename = path.filename();
 		spdlog::info("LoadStreamBsp: {}", filename.string());
 
 		// resolve modded stbsp path so we can load mod stbsps
 		auto modFile = g_pModManager->m_ModFiles.find(g_pModManager->NormaliseModFilePath(fs::path("maps" / filename)));
 		if (modFile != g_pModManager->m_ModFiles.end())
 		{
-			// need to allocate a new string for this
-			std::string newPath = (modFile->second.m_pOwningMod->m_ModDirectory / "mod" / modFile->second.m_Path).string();
-			allocatedNewPath = new char[newPath.size() + 1];
-			strncpy_s(allocatedNewPath, newPath.size() + 1, newPath.c_str(), newPath.size());
-			pPath = allocatedNewPath;
+			newPath = (modFile->second.m_pOwningMod->m_ModDirectory / "mod" / modFile->second.m_Path).string();
+			pPath = newPath.c_str();
 		}
 	}
+	else if (path.extension() == ".starpak")
+	{
+		// code for this is mostly stolen from above
+
+		// unfortunately I can't find a way to get the rpak that is causing this function call, so I have to
+		// store them on mod init and then compare the current path with the stored paths
+
+		// game adds r2\ to every path, so assume that a starpak path that begins with r2\paks\ is a vanilla one
+		// modded starpaks will be in the mod's paks folder
+		if (*++path.begin() != "paks")
```

Comment:
this is a nice trick but not super readable, would be better to use something more immediately readable here